### PR TITLE
Add naming guidance for blockquote, meter and time roles

### LINKED ISSUE #1077: Update section on providing accessible names and descriptions for ARIA 1.2
* Add guidance on new label, legend, and fieldset roles.
* Names are prohibitted on caption, code, deletion, emphasis, insertion, paragraph, presentation, strong, subscript, superscript.
* Names will also be prohibitted on generic role when it is added to ARIA 1.2.

### LINKED ISSUE #1212: Update Naming and Describing section for new naming mechanisms
Add guidance on new label, legend, and fieldset roles.

( split from #1077  )

### PR DESCRIPTION
**Pull request by zcorpan**
Fixes #1077.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/pull/1231.html" title="Last updated on Nov 6, 2019, 4:30 AM UTC (07a57d4)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria-practices/1231/6ec616f...07a57d4.html" title="Last updated on Nov 6, 2019, 4:30 AM UTC (07a57d4)">Diff</a>

### CODE REVIEW

File: aria-practices.html [hunk 1]
```diff
@@ -4869,6 +4890,16 @@ <h4><span id="naming_role_guidance_heading">Accessible Name Guidance by Role</sp
                 </ul>
               </td>
             </tr>
+            <tr>
+              <td><a href="#blockquote" class="role-reference"><code>time</code></a></td>
```
Comment (mcking65) [inline aria-practices.html]: Fix href:

```suggestion
              <td><a href="#time" class="role-reference"><code>time</code></a></td>
```

File: aria-practices.html [hunk 2]
```diff
@@ -4869,6 +4890,16 @@ <h4><span id="naming_role_guidance_heading">Accessible Name Guidance by Role</sp
...
+              <td>Discretionary</td>
+              <td>
+                <ul>
+                  <li>Naming is optional, but can potentially help screen reader users understand the context and purpose of the given specific point in time.</li>
```
[the remark this answers is not in the document]
Comment (zcorpan) [reply aria-practices.html]: I'm not sure if naming a `time` is ever useful. A `time`, from what I understand, isn't typically focusable, but is more like a normal word within a paragraph. To understand the context, you'd read the full paragraph.

Looking at the examples of `<time>` in HTML: https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-time-element

Should we say "Do Not Name" instead?

[the remark this answers is not in the document]
Comment (zcorpan) [reply aria-practices.html]: Do Not Name - not supported by assistive technologies.

File: aria-practices.html [hunk 3]
```diff
@@ -4235,6 +4235,16 @@ <h4><span id="naming_role_guidance_heading">Accessible Name Guidance by Role</sp
                 </ul>
               </td>
             </tr>
+            <tr>
+              <td><a href="#blockquote" class="role-reference"><code>blockquote</code></a></td>
+              <td>Discretionary</td>
+              <td>
+                <ul>
+                  <li>Naming is optional, but can potentially help screen reader users understand the context and purpose of the quote.</li>
+                  <li>Named using <code>aria-labelledby</code> if a visible label is present, otherwise with <code>aria-label</code>.</li>
```
Comment (zcorpan) [inline aria-practices.html]: Discourage aria-label: discretionary is kinda positive. This wording may encourage people to label all their blockquotes. Doesn't improve the experience.

File: aria-practices.html [hunk 4]
```diff
@@ -4235,6 +4235,16 @@ <h4><span id="naming_role_guidance_heading">Accessible Name Guidance by Role</sp
                 </ul>
               </td>
             </tr>
+            <tr>
+              <td><a href="#blockquote" class="role-reference"><code>blockquote</code></a></td>
+              <td>Discretionary</td>
+              <td>
+                <ul>
+                  <li>Naming is optional, but can potentially help screen reader users understand the context and purpose of the quote.</li>
```
Comment (zcorpan) [inline aria-practices.html]: Remove this bullet point.

File: aria-practices.html (hunk 3 above)
Comment (zcorpan) [reply aria-practices.html]: "If a visible label is present, associating it with the blockquote by using aria-labelledby could benefit some AT users."

**Comment by jongund** [comment]
@mcking65 In section 1.8 Heading Role, what if someone wants to use a `aria-level` value greater than 6?   For example, in some type of document with a high level of nesting.  Is that going to be allowed?  If not we should state the allowed range i 1 to 6, if allowed there should eb some type of guidance on values greater than 6.

**Comment by jongund** [comment]
@mcking65 In section 8.2 for `listitem` role, wouldn't a better example be for lists that are partially rendered list do to size/length of the list and or bandwidth limitations of the device.   This I believe this is one of the primary reason for other similar attributes like `aria-posinset` and `aria-setsize` to be included in ARIA.

**Comment by carmacleod** [comment]
@jongund I think your 2 comments above probably belong in PR https://github.com/w3c/aria-practices/pull/1109, which is about `aria-level`.

File: aria-practices.html [hunk 5]
```diff
@@ -4869,6 +4887,11 @@ <h4><span id="naming_role_guidance_heading">Accessible Name Guidance by Role</sp
                 </ul>
               </td>
             </tr>
+            <tr>
+              <td><a href="#time" class="role-reference"><code>time</code></a></td>
+              <td>Do Not Name</td>
+              <td>Not supported by assistive technologies.</td>
```
Comment (carmacleod) [inline aria-practices.html]: Consider changing:
> Not supported by assistive technologies.

to:
> Naming not supported by assistive technologies.

... just to ensure that authors don't accidentally get the impression that the _time role_ itself is not supported by AT...

Comment (zcorpan) [reply aria-practices.html]: Thanks, fixed.

**Comment by zcorpan** [comment]
@mcking65 I've filed https://github.com/w3c/aria/issues/1117 on ARIA.

**Pull request merged**